Concentration/ViewController.swift

```diff
@@ -9,58 +9,104 @@ import UIKit
+        flipCountLabel.text = "Flips: \(game.flipCount)"
+        for index in cardButtons.indices { // making sure every card is viewed correctly
             let button = cardButtons[index]
             let card = game.cards[index]
             if card.isFaceUp {
                 button.setTitle(emoji(for: card), for: UIControl.State.normal)
                 button.backgroundColor = #colorLiteral(red: 1.0, green: 1.0, blue: 1.0, alpha: 1.0)
             } else {
                 button.setTitle("", for: UIControl.State.normal)
-                button.backgroundColor = card.isMatched ? #colorLiteral(red: 1, green: 1, blue: 1, alpha: 0) : #colorLiteral(red: 1, green: 0.5781051517, blue: 0, alpha: 1)
+                button.backgroundColor = card.isMatched ? #colorLiteral(red: 1, green: 1, blue: 1, alpha: 0) : themeColors[emojiChoices.key]!.0
             }
         }
     }
     
-    var emojiChoices = ["🦇", "😱", "🙀", "😈", "🎃", "👻", "🍭", "🍬", "🍎"]
+    // A dictionary mapping themes (Strings) to an array of emoji choices (array of Strings)
+    let themesEmojies = ["Halloween": ["🦇", "😱", "🙀", "😈", "🎃", "👻", "🍭", "🍬", "🍎"],
+                  "Animals": ["🙀", "🦇", "🐱", "🐥", "🐦", "🦉", "🐺", "🦄", "🐷" ],
+                  "Sports": ["⚽️", "🥎", "⚾️", "🥋", "🎽", "🥏", "🤿","🏒", "🏉" ],
+                  "Food": ["🍭", "🎃", "🍎", "🍬", "🍌", "🥝" ,"🍞", "🥕", "🥯"],
+                  "Faces": ["😱", "😀", "😋", "🥸", "😜", "🥳", "😛", "🤩", "😏"],
+                  "Flags": ["🏳️‍🌈", "🏳️", "🏴","🏴‍☠️","🏁","🇧🇿","🇫🇯","🇹🇭","🇮🇳"]
+                ]
+    
+    // A dictionary mapping themes (Strings) to tuples of color literals, where the first
+    // color is for the background (bg) of the cards, and the second is for the overall bg.
+    let themeColors = ["Halloween": (#colorLiteral(red: 1, green: 0.5781051517, blue: 0, alpha: 1), #colorLiteral(red: 0, green: 0, blue: 0, alpha: 1)),
+                       "Animals": (#colorLiteral(red: 0.4666666687, green: 0.7647058964, blue: 0.2666666806, alpha: 1), #colorLiteral(red: 0.7254902124, green: 0.4784313738, blue: 0.09803921729, alpha: 1)),
+                       "Sports": (#colorLiteral(red: 0.2196078449, green: 0.007843137719, blue: 0.8549019694, alpha: 1), #colorLiteral(red: 0.9764705896, green: 0.850980401, blue: 0.5490196347, alpha: 1)),
+                       "Food": (#colorLiteral(red: 0.8549019694, green: 0.250980407, blue: 0.4784313738, alpha: 1), #colorLiteral(red: 0.1764705926, green: 0.4980392158, blue: 0.7568627596, alpha: 1)),
+                       "Faces": (#colorLiteral(red: 0.6000000238, green: 0.6000000238, blue: 0.6000000238, alpha: 1), #colorLiteral(red: 0.8039215803, green: 0.8039215803, blue: 0.8039215803, alpha: 1)),
+                       "Flags": (#colorLiteral(red: 0, green: 0, blue: 0, alpha: 1), #colorLiteral(red: 0.2588235438, green: 0.7568627596, blue: 0.9686274529, alpha: 1))
+                     ]
+    
+    lazy var emojiChoices = themesEmojies.randomElement()!
```

Comment:
Can you explain why it is `lazy`? :)

Reply:
I think so. It is lazy because it depends on another property, and it is impossible to define properties this way (through other properties). This way (using lazy), themeEmojies is initialized when ViewController is initialized, and emojiChoices is only initialized when is used for the first time (and by that time ViewController is already initialized, therefore themesEmojies is initialized/defined as well).